StripeCore/StripeCore/Source/Categories/Locale+StripeCore.swift

```diff
@@ -41,4 +41,54 @@ import Foundation
         return self.isoRegionCodes
 #endif
     }
+
+    /// Returns the BCP 47(-ish) language tag representing the locale.
+    ///
+    /// The language tag is expected to be well-formed as log as the locale identifier contains a
+    /// valid language code. For example:
+    ///
+    /// ```
+    /// let locale = Locale(identifier: "fr_CA")
+    /// locale.toLanguageTag() // -> "fr-CA"
+    /// ```
+    ///
+    /// The following example returns `"-ES"`, even though `"und-ES"` will be the appropriate BCP 47 tag:
+    ///
+    /// ```
+    /// let locale = Locale(identifier: "_ES")
+    /// locale.toLanguageTag() // -> "-ES"
+    /// ```
+    /// All system iOS and macOS locales are expected to contain valid language codes.
+    ///
+    /// On iOS 16+, the device region may be different from the language region. When these are different,
+    /// the device region is encoded at the end. The example below corresponds to:
+    /// Language=English (UK) and Region=United States:
+    ///
+    /// ```
+    /// let locale = Locale(identifier: "en_GB@rg=uszzzz")
+    /// locale.toLanguageTag() // -> "en-GB"
+    /// ```
+    ///
+    func toLanguageTag() -> String {
+        var tag = Locale.canonicalLanguageIdentifier(from: self.identifier)
+
+        // Drop sub-tags or extended variants like `en-US@calendar=gregorian`
+        // or `en-GB@rg=uszzzz`
+        if let unextended = tag.split(separator: "@").first {
+            tag = String(unextended)
+        }
```

Comment:
I initially thought that we could just use `Locale.identifier(.bcp47, from: self.identifier)` ([ref](https://developer.apple.com/documentation/foundation/locale/4020220-identifier)) on iOS 16, but this ends up dropping the script for `zh-Hant-TW` -> `zh_TW` and `zh-Hans-CN` -> `zh_CN` making the problem worse. Since we don't ever use the extended characters features in the locale, I think truncating it at the `@` character seemed like more scalable solution.